miden-crypto/benches/hash.rs

```diff
@@ -100,6 +100,52 @@ fn rpx256_sequential(c: &mut Criterion) {
     });
 }
 
+fn poseidon2_2to1(c: &mut Criterion) {
+    let v: [Word; 2] = [Poseidon2::hash(&[1_u8]), Poseidon2::hash(&[2_u8])];
+    c.bench_function("Poseidon2 2-to-1 hashing (cached)", |bench| {
+        bench.iter(|| Poseidon2::merge(black_box(&v)))
```

Comment:
How does this implementation compare with HorizenLabs'?

Reply:
I am getting `7.7344 µs` for their version vs `1.7 µs` in this PR for 1 call to the permutation. Probably it is due to their use of Arkworks generic finite field arithmetic, but not sure.